[ML] Update auditbeat modules for ECS

sophiec20 opened the pull request:
## Summary

Updates the fields used in the auditbeat hosts and docker data recognizer modules to the Elastic Common Schema (ECS) field names.

### Checklist

- [x] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)

### For maintainers

- [x] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

sophiec20 committed:
[ML] Initial commit for auditbeat hosts ECS

Rename fields for ECS
Rework dashboards due to bwc

sophiec20 committed:
[ML] Further auditbeat tidy up and consistency changes

Custom urls should link to saved search, not discover
Ensure savedSearchId is used for visualizations
Ensure filter terms are consistent

TODO Decide if we should rename to auditd module
TODO Fix for new saved object format

elasticmachine commented:
Pinging @elastic/ml-ui

sophiec20 committed:
[ML] Refinements for auditbeat host module

Remove duplicated title from visState
Shrink panel heights in row 1

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/10632/)

sophiec20 committed:
[ML] Refinements to auditbeat module

Update module name from auditd to auditbeat
Add useMargins true for dashboards
Add filter to custom url for
   exists auditd.data.syscall
   not exists container.runtime
   event.module: auditd

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/10778/)

sophiec20 committed:
[ML] Initial commit for auditbeat_process_docker_ecs

Update for ECS using
  container.name (instead of container.id)
  container.runtime: docker
  process.executable
  event.module: auditd
  auditd.data.syscall exists

TODOs
Use auditd.message_type: syscall (instead of auditd.data.syscall)
Possibly combine with auditbeat hosts saved objects (depending on host.name being shared)
Possibly combine to single dashboard
Test against live auditbeat data collection

sophiec20 commented:
TODO change `bucket_span` back to 1h ... currently set to 10m due to limited test data

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/10957/)

peteharverson commented on x-pack/plugins/ml/server/models/data_recognizer/modules/auditbeat_process_docker_ecs/ml/docker_high_count_process_events_ecs.json:
```diff
@@ -0,0 +1,41 @@
+    "bucket_span": "10m",
+    "detectors": [
+      {
+        "detector_description": "high_non_zero_count partition container.name",
```
I know this is the same as the current non ECS module, but I wonder if we should give this detector a more user-friendly description? `High process execution rate` ?

peteharverson commented on x-pack/plugins/ml/server/models/data_recognizer/modules/auditbeat_process_docker_ecs/ml/docker_rare_process_activity_ecs.json:
```diff
@@ -0,0 +1,40 @@
+      {
+        "function": "rare",
+        "by_field_name": "process.executable",
+        "partition_field_name": "container.name"
```
I think we should give this a user-friendly detector description. `rare process execution` maybe?

peteharverson commented on x-pack/plugins/ml/server/models/data_recognizer/modules/auditbeat_process_hosts_ecs/ml/hosts_high_count_process_events_ecs.json:
```diff
@@ -0,0 +1,41 @@
+    "bucket_span": "10m",
+    "detectors": [
+      {
+        "detector_description": "high_non_zero_count partition host.name",
```
As above, maybe this should be something like `High process execution rate`?

peteharverson commented on x-pack/plugins/ml/server/models/data_recognizer/modules/auditbeat_process_hosts_ecs/ml/hosts_rare_process_activity_ecs.json:
```diff
@@ -0,0 +1,41 @@
+      {
+        "function": "rare",
+        "by_field_name": "process.executable",
+        "partition_field_name": "host.name"
```
As above, maybe give this a user-friendly detector description like `rare process execution` or even just `rare process`?

peteharverson reviewed (commented):
This looks good, although we should maybe take the opportunity to give the detectors more user-friendly descriptions?

peteharverson reviewed (approved):
LGTM - will edit bucket_span to 1hr and add user-friendly detector descriptions in a follow up PR.
